File: aditus_backend_service/app/models/device.py
from datetime import datetime
from app import db

class Device(db.Model):
    __tablename__ = 'devices'

    id = db.Column(db.Integer, primary_key=True)
    owner_id = db.Column(db.Integer, db.ForeignKey('users.id'), nullable=False, index=True)
    name = db.Column(db.String(128), unique=False, nullable=False)
    public_key = db.Column(db.String, unique=True, nullable=False, index=True)
    created_at = db.Column(db.DateTime, default=datetime.now, nullable=False)
    last_used_at = db.Column(db.DateTime, default=datetime.now, onupdate=datetime.now, nullable=False)

    # Relationships
    owner = db.relationship('User', back_populates='devices')
    access_logs = db.relationship('AccessLog', back_populates='device', lazy='dynamic')

    def to_dict(self, include_owner=False):
        """Convert device to dictionary"""
        data = {
            'id': self.id,
            'owner_id': self.owner_id,
            'name': self.name,
            'public_key': self.public_key,
            'created_at': self.created_at.isoformat() if self.created_at else None,
            'last_used_at': self.last_used_at.isoformat() if self.last_used_at else None,
        }

        if include_owner and self.owner:
            data['owner'] = self.owner.to_dict()

        return data

    def __repr__(self):
        return f'<Device {self.name} (owner: {self.owner_id})>'

File: smartphone_client_app/lib/features/device/data/models/device.dart
import 'package:equatable/equatable.dart';

class Device extends Equatable {
  final int id;
  final int? ownerId;
  final String name;
  final String publicKey;
  final DateTime? createdAt;
  final DateTime? lastUsedAt;

  const Device({
    required this.id,
    this.ownerId,
    required this.name,
    required this.publicKey,
    this.createdAt,
    this.lastUsedAt,
  });

  factory Device.fromJson(Map<String, dynamic> json) {
    return Device(
      id: json['id'] as int,
      ownerId: json['owner_id'] as int?,
      name: json['name'] as String,
      publicKey: json['public_key'] as String,
      createdAt: json['created_at'] != null
          ? DateTime.parse(json['created_at'] as String)
          : null,
      lastUsedAt: json['last_used_at'] != null
          ? DateTime.parse(json['last_used_at'] as String)
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      if (ownerId != null) 'owner_id': ownerId,
      'name': name,
      'public_key': publicKey,
      if (createdAt != null) 'created_at': createdAt!.toIso8601String(),
      if (lastUsedAt != null) 'last_used_at': lastUsedAt!.toIso8601String(),
    };
  }

  @override
  List<Object?> get props => [
    id,
    ownerId,
    name,
    publicKey,
    createdAt,
    lastUsedAt,
  ];
}
